src/views/AddTimerView.tsx

```diff
@@ -0,0 +1,153 @@
+import type React from 'react';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useTimerContext } from '../context/TimerContext';
+
+const AddTimerView: React.FC = () => {
+    const { addTimer } = useTimerContext();
+    const navigate = useNavigate();
+
+    // State to manage timer type and configuration
+    const [type, setType] = useState<'countdown' | 'xy' | 'tabata' | 'stopwatch'>('countdown');
+    const [config, setConfig] = useState<any>({});
+    const [error, setError] = useState<string | null>(null);
+
+    // Validate and handle adding a new timer
+    const handleAddTimer = () => {
+        if (
+            (type === 'countdown' && !config.totalSeconds) ||
+            (type === 'xy' && (!config.timePerRound || !config.totalRounds)) ||
+            (type === 'tabata' && (!config.workSeconds || !config.restSeconds || !config.totalRounds)) ||
+            (type === 'stopwatch' && !config.totalSeconds)
+        ) {
+            setError('Please fill in all required fields for the selected timer type.');
+            return;
+        }
+
+        const newTimer = {
+            id: `${type}-${Date.now()}`,
+            type,
+            config,
+            state: 'not running' as const,
+        };
+
+        addTimer(newTimer);
+        navigate('/');
+    };
+
+    // Dynamically render input fields based on the timer type
+    const renderConfigFields = () => {
```

Comment:
Same as other comment - you should put this logic into a separate component, rather than a nested function.